### PR TITLE
[DDPB-3951] behat rewrite for admin app notifications

### PR DESCRIPTION
**Pull request by coderste**
## Purpose
behat rewrite for admin app notifications

Fixes DDPB-3951

### CODE REVIEW

File: api/tests/Behat/behat.yml [hunk 1]
```diff
@@ -238,6 +238,12 @@ v2-tests-goutte:
             contexts:
                 - App\Tests\Behat\v2\OrganisationManagement\OrganisationManagementFeatureContext
 
+        app-notification:
+            description: Coverage of organisation management features - adding, deleting, editing
```
Comment (acsauk) [inline api/tests/Behat/behat.yml:242]: This needs updating I think 😄

File: api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+namespace App\Tests\Behat\v2\AppNotification;
+
+trait AppNotificationTrait
+{
+    private string $validationMsg = 'Please enter a message';
+
+    /**
+     * @When I visit the service notification page
+     */
+    public function iNavigateToTheClientsSearchPage()
```
Comment (acsauk) [inline api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php:14]: This also needs updating 😹

File: api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        $this->iAmOnAdminNotificationPage();
+
+        $this->selectOption('setting[enabled]', '0');
+        $this->pressButton('setting[save]');
+
+        $this->iVisitTheClientLoginPage();
+        $this->iAmOnClientLoginPage();
+
+        $el = $this->getSession()->getPage()->find('css', '.behat-region-service-notification');
+        assert(null == $el);
```
Comment (acsauk) [inline api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php:84]: Probably best to stay consistent with our assert calls - `AssertTrait::assertIsNull()` should work here

File: api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    /**
+     * @Then I should see the service message on the client login page
+     */
+    public function iShouldSeeTheServiceMessageOnTheClientLoginPage()
+    {
+        $this->iVisitTheClientLoginPage();
+        $this->iAmOnClientLoginPage();
+
+        $notification = $this->getSession()->getPage()->find('css', '.behat-region-service-notification > .opg-alert > .behat-region-alert-message > p')->getText();
+        assert('Lorem ipsum' == $notification);
```
Comment (acsauk) [inline api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php:65]: You can use AssertTrait::assertStringEqualsString() here (for consistency sake - it does the same thing).

File: api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        $this->fillField('setting[content]', 'Lorem ipsum');
+        $this->selectOption('setting[enabled]', '1');
+        $this->pressButton('setting[save]');
+
+        $this->iVisitTheClientLoginPage();
+        $this->iAmOnClientLoginPage();
+
+        $notification = $this->getSession()->getPage()->find('css', '.behat-region-service-notification > .opg-alert > .behat-region-alert-message > p')->getText();
+        assert('Lorem ipsum' == $notification);
```
Comment (acsauk) [inline api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php:43]: As per others below - `AssertTrait::assertStringEqualsString()` would be good here for consistency.

**Review by acsauk — CHANGES_REQUESTED** [review]
Looking good - couple of small changes then ready to merge 💪

File: api/tests/Behat/behat.yml (hunk 1 above)
Comment (coderste) [reply api/tests/Behat/behat.yml:242]: Indeed it does

File: api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php (hunk 2 above)
Comment (coderste) [reply api/tests/Behat/bootstrap/v2/AppNotification/AppNotificationTrait.php:14]: Yep 😆

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**